src/plugins/event-plugins/FetchPlugin.ts

```diff
@@ -85,7 +85,7 @@ export class FetchPlugin extends MonkeyPatched implements Plugin {
     };
 
     private isSessionRecorded = () => {
-        return this.context.getSession().record;
+        return this.context.getSession()?.record || false;
```

Comment:
Please add the following test to `FetchPlugin.test.ts` to verify the new behavior and prevent regressions:

```
    test('when getSession returns undefined then the plugin does not record a trace', async () => {
        const getSession: jest.MockedFunction<GetSession> = jest.fn(
            () => undefined
        );
        const config: PartialHttpPluginConfig = {
            logicalServiceName: 'sample.rum.aws.amazon.com',
            urlsToInclude: [/aws\.amazon\.com/]
        };
        const xRayOnContext: PluginContext = {
            applicationId: 'b',
            applicationVersion: '1.0',
            config: { ...DEFAULT_CONFIG, ...{ enableXRay: true } },
            record,
            recordPageView,
            getSession
        };

        const plugin: FetchPlugin = new FetchPlugin(config);
        plugin.load(xRayOnContext);

        // Run
        await fetch('https://aws.amazon.com');
        plugin.disable();

        // Assert
        expect(mockFetch).toHaveBeenCalledTimes(1);
        expect(record).not.toHaveBeenCalled();
    });
```